modules/apps/portal/portal-user-locale-options/src/main/resources/META-INF/resources/dynamic_include/user_locale_options.jsp

```diff
@@ -0,0 +1,58 @@
+<%--
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+--%>
+
+<%@ include file="/dynamic_include/init.jsp" %>
+
+<%
+Locale userLocale = user.getLocale();
+%>
+
+<liferay-util:buffer
+	var="alertMessage"
+>
+	<div dir="<%= LanguageUtil.get(userLocale, "lang.dir") %>">
+		<div class="d-block">
+			<%= LanguageUtil.format(userLocale, "this-page-is-displayed-in-x", locale.getDisplayName(userLocale)) %>
+		</div>
+
+		<c:if test="<%= LanguageUtil.isAvailableLocale(themeDisplay.getSiteGroupId(), user.getLocale()) %>">
+			<aui:a cssClass="d-block" href='<%= themeDisplay.getPathMain() + "/portal/update_language?p_l_id=" + themeDisplay.getPlid() + "&redirect=" + URLCodec.encodeURL(currentURL) + "&languageId=" + user.getLanguageId() + "&persistState=false&showUserLocaleOptionsMessage=false" %>'>
+				<%= LanguageUtil.format(userLocale, "display-the-page-in-x", userLocale.getDisplayName(userLocale)) %>
+			</aui:a>
+		</c:if>
+	</div>
+
+	<div dir="<%= LanguageUtil.get(request, "lang.dir") %>">
+		<aui:a cssClass="d-block" href='<%= themeDisplay.getPathMain() + "/portal/update_language?p_l_id=" + themeDisplay.getPlid() + "&redirect=" + URLCodec.encodeURL(currentURL) + "&languageId=" + themeDisplay.getLanguageId() + "&showUserLocaleOptionsMessage=false" %>'>
+			<%= LanguageUtil.format(locale, "set-x-as-your-preferred-language", locale.getDisplayName(locale)) %>
+		</aui:a>
+	</div>
+</liferay-util:buffer>
+
+<aui:script>
+	Liferay.Util.openToast({
+		message: '<%= HtmlUtil.escapeJS(alertMessage) %>',
+		messageType: 'html',
+		onClose: function (data) {
+			if (data.event) {
```

Comment:
If we want, we can simply pass `event` here... that's a common enough pattern, but I think we're trying to favour destructuring here, so in general we'll pass complex objects and destructure them rather than use positional arguments?

This makes it easier to change APIs :)